### ISSUE TITLE
Exception: Sequence contains more than one element

### ISSUE THREAD

**Issue by skrysmanski** [microsoft/vstest#2926]
## Description

In version 2.2.3 there is a regression (compared to 2.1.2) for this code:

```c#
[TestClass]
public sealed class MyTestClass : MyTestClassBase
{
    [TestMethod]
    [TestCategory("MyTests")]
    public new void TestSomething()
    {
        base.TestSomething();
    }
}

public abstract class MyTestClassBase
{
    protected void TestSomething()
    {
        // Do something
    }
}
```

Executing this via:

```
"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe" "MyTests.dll" /TestCaseFilter:"TestCategory=MyTests"
```

Leads to this error (I had to manually translate it because the error is in German on my machine):

```
Exception in call of Executor "executor://mstestadapter/v2": The sequence contains more than one element.
```

This does not happen in version 2.1.2.

## Steps to reproduce

1. Create a modern-SDK test project
1. Select `net48` as target framework
1. Add code above
1. Reference NuGet packages like this:

```
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.9.1" />
    <PackageReference Include="MSTest.TestAdapter" Version="2.2.3" />
    <PackageReference Include="MSTest.TestFramework" Version="2.2.3" />
```

Run the tests via `vstest.console` (like shown above).

## Expected behavior

There should be no exception.

## Actual behavior

The exception above is thrown.

## Environment

* Windows 10
* Visual Studio 2019 16.7.12
* vstest.console 16.7.0

[AB#1335565](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1335565)

**Comment by skrysmanski** [comment]
The strange thing is that it's working with `MSTest.TestFramework`/`MSTest.TestFramework`  in version 2.1.2 - it's just broken in version 2.2.x. But I guess they've changed something in the test framework, right?

**Comment by Haplois** [comment]
Yes, we switched to a more robust test resolving mechanism provided by the Test Platform v16.10, but it doesn't handle hidden methods well. That needs to be fixed for this to work. 

For now, as a workaround, you can make the base method and override it if possible, or you can rename your test.

**Comment by skrysmanski** [comment]
> This will be fixed in our 16.11 release.

So, now that 16.11 is released, I was trying to see whether the bug is finally fixed. But it's not yet fixed. Did the fix not make it into the 16.11 release?

**Comment by skrysmanski** [comment]
Just as a side note: Apparently some change in either the "MSTest.TestAdapter" or the "MSTest.TestFramework" package in **version 2.2.4** changed the behavior so that we now get a warning rather than an error. Unfortunately, this also means that tests that run into this problem are being "lost" because a CI pipeline can no longer detect this error.

**Comment by deWisch** [comment]
Any news on this issue? It is still present in the latest version.

**Comment by Naxemar** [comment]
Any news ?

**Comment by skrysmanski** [comment]
This issue is finally fixed with version 3.0.2. Thanks :)
